tsl/src/bgw_policy/move_hyper_inval_api.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+
+#include <postgres.h>
+
+#include "move_hyper_inval_api.h"
+
+#include "bgw/job.h"
+#include "bgw/job_stat.h"
+#include "bgw/timer.h"
+#include "bgw_policy/job.h"
+#include "bgw_policy/job_api.h"
+#include "guc.h"
+#include "hypertable.h"
+#include "jsonb_utils.h"
+#include "policy_config.h"
+#include <utils/elog.h>
+
+#define DEFAULT_MAX_RUNTIME                                                                        \
+	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("0"), InvalidOid, -1))
+
+Datum
+policy_move_hyper_inval_proc(PG_FUNCTION_ARGS)
+{
+	if (PG_NARGS() != 2 || PG_ARGISNULL(0) || PG_ARGISNULL(1))
+		PG_RETURN_VOID();
+
+	ts_feature_flag_check(FEATURE_POLICY);
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+	policy_move_hyper_inval_execute(PG_GETARG_INT32(0), PG_GETARG_JSONB_P(1));
+
+	PG_RETURN_VOID();
+}
+
+Datum
+policy_move_hyper_inval_check(PG_FUNCTION_ARGS)
+{
+	if (PG_ARGISNULL(0))
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_NULL_VALUE_NOT_ALLOWED), errmsg("config must not be NULL")));
+	}
+
+	policy_move_hyper_inval_read_and_validate_config(PG_GETARG_JSONB_P(0), NULL);
+
+	PG_RETURN_VOID();
+}
+
+static int32
+policy_move_hyper_inval_add_internal(Hypertable *ht, bool if_not_exists,
+									 Interval *schedule_interval, Oid owner_id,
+									 TimestampTz initial_start, const bool fixed_schedule,
+									 char *timezone, const NullableDatum *include_tiered_data)
+{
+	NameData application_name, proc_name, proc_schema, check_schema, check_name, owner;
+
+	List *jobs = ts_bgw_job_find_by_proc_and_hypertable_id(POLICY_MOVE_HYPER_INVAL_PROC_NAME,
+														   FUNCTIONS_SCHEMA_NAME,
+														   ht->fd.id);
+	if (jobs != NIL)
+	{
+		Assert(list_length(jobs) == 1);
+		if (!if_not_exists)
+			ereport(ERROR,
+					errcode(ERRCODE_DUPLICATE_OBJECT),
+					errmsg("move hypertable invalidations policy already exists for \"%s\"",
+						   get_rel_name(ht->main_table_relid)));
+		else
+			ereport(NOTICE,
+					errmsg("move hypertable invalidations policy already exists for \"%s\", "
+						   "skipping",
+						   get_rel_name(ht->main_table_relid)));
+		return -1;
+	}
+
+	namestrcpy(&application_name, "Move Hypertables Invalidation Policy");
+	namestrcpy(&proc_name, POLICY_MOVE_HYPER_INVAL_PROC_NAME);
+	namestrcpy(&proc_schema, FUNCTIONS_SCHEMA_NAME);
+	namestrcpy(&check_name, POLICY_MOVE_HYPER_INVAL_CHECK_NAME);
+	namestrcpy(&check_schema, FUNCTIONS_SCHEMA_NAME);
+	namestrcpy(&owner, GetUserNameFromId(owner_id, false));
+
+	JsonbParseState *parse_state = NULL;
+
+	pushJsonbValue(&parse_state, WJB_BEGIN_OBJECT, NULL);
+	ts_jsonb_add_int32(parse_state, POLICY_CONFIG_KEY_HYPERTABLE_ID, ht->fd.id);
+
+	if (!include_tiered_data->isnull)
+		ts_jsonb_add_bool(parse_state,
```

Comment:
Again, it is unclear what any of this has to do with data tiering? Is it maybe something just copied over from the refresh policy that has no applicability here?

Otherwise, please explain in a comment and PR description what this is for.